data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -520,6 +534,9 @@ public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
                 ownedPartitionsEpoch.remove(topicPartition);
                 partitionCommitTrackerMap.remove(topicPartition.partition());
             }
+            if (paused) {
```

Comment:
is it really required in partition revocation?

Reply:
Probably not. I thought about it before adding this code. I do not see any down side of doing it. Let me know if you can think of any side effects of this code.